deepchem/models/jax_models/jax_model.py

```diff
@@ -0,0 +1,373 @@
+
+import warnings
+
+logger = logging.getLogger(__name__)
+
+
+class JaxModel(Model):
+  """This is a DeepChem model implemented by a Jax Model
+
+  Here is a simple example of that uses JaxModel to train a
+  Haiku (JAX Neural Network Library) based model on deepchem
+  dataset.
+
+  >> def f(x):
+  >>   net = hk.nets.MLP([512, 256, 128, 1])
+  >>   return net(x)
+  >> model = hk.without_apply_rng(hk.transform(f))
+  >> rng = jax.random.PRNGKey(500)
+  >> x, _, _, _ = next(iter(train_dataset.iterbatches(batch_size=256)))
+  >> params = model.init(rng, x)
+  >> j_m = JaxModel(model, params, 256, 0.001, 100)
+  >> j_m.fit(train_dataset)
+
+  All optimizations will be done using the optax library.
+  """
+
+  def __init__(self,
+               model,
+               params: hk.Params,
+               loss: Union[Loss, LossFn],
+               output_types: Optional[List[str]] = None,
+               batch_size: int = 100,
+               learning_rate: float = 0.001,
+               optimizer: Union[optax.GradientTransformation,
+                                Optimizer] = optax.adam(1e-3),
+               log_frequency: int = 100,
+               **kwargs):
+    """
+    Create a new JaxModel
+
+    Parameters
+    ----------
+    model: hk.State or Function
+      Any Jax based model that has a `apply` method for computing the network.
+    params: hk.Params
+      The parameter of the Jax based networks
+    loss: dc.models.losses.Loss or function
+      a Loss or function defining how to compute the training loss for each
+      batch, as described above
+    output_types: list of strings, optional (default None)
+      the type of each output from the model, as described above
+    batch_size: int, optional (default 100)
+      default batch size for training and evaluating
+    learning_rate: float or LearningRateSchedule, optional (default 0.001)
+      the learning rate to use for fitting.  If optimizer is specified, this is
+      ignored.
+    optimizer: optax object
+      For the time being, it is optax object
+    log_frequency: int, optional (default 100)
+      The frequency at which to log data. Data is logged using
+      `logging` by default.
+
+    Miscellanous Parameters Yet To Add
+    ----------------------------------
+    model_dir: str, optional (default None)
+      Will be added along with the save & load method
+    tensorboard: bool, optional (default False)
+      whether to log progress to TensorBoard during training
+    wandb: bool, optional (default False)
+      whether to log progress to Weights & Biases during training
+
+    Work in Progress
+    ----------------
+    [1] Integerate the optax losses, optimizers, schedulers with Deepchem
+    [2] Support for saving & loading the model.
+    """
+    super(JaxModel, self).__init__(model=model, **kwargs)
+    warnings.warn(
+        'JaxModel is still under work and hence a lot of the method might not be implemented'
```

Comment:
How about "JaxModel is still in active development and all features may not yet be implemented"